snuba/subscriptions/worker.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import annotations
+
+from typing import NamedTuple, Mapping, MutableSequence, Optional, Sequence
+
+from snuba.subscriptions.consumer import Tick
+from snuba.subscriptions.data import Subscription
+from snuba.subscriptions.executor import SubscriptionExecutor
+from snuba.subscriptions.scheduler import ScheduledTask, Scheduler
+from snuba.utils.streams.batching import AbstractBatchWorker
+from snuba.utils.streams.producer import Producer
+from snuba.utils.streams.types import Message, Topic
+from snuba.web.query import ClickhouseQueryResult
+
+
+class SubscriptionResult(NamedTuple):
+    task: ScheduledTask[Subscription]
+    result: ClickhouseQueryResult
+
+
+class SubscriptionWorker(AbstractBatchWorker[Tick, Sequence[SubscriptionResult]]):
+    def __init__(
+        self,
+        executor: SubscriptionExecutor,
+        schedulers: Mapping[int, Scheduler[Subscription]],
```

Comment:
Should `int` be `PartitionId`?

Reply:
This comes from the batching consumer, so it doesn't need to be. Ultimately, I'd like to make the key in this mapping a `Partition` instead (and in other places, see https://github.com/getsentry/snuba/pull/746#discussion_r374292578.)